tests/Functional/Query/QueryBuilderTest.php

```diff
@@ -380,4 +496,16 @@ private function platformSupportsSkipLocked(): bool
 
         return ! $platform instanceof SQLitePlatform;
     }
+
+    private function platformSupportsCTEs(): bool
+    {
+        $platform = $this->connection->getDatabasePlatform();
+
+        return ! $platform instanceof MySQLPlatform || $platform instanceof MySQL80Platform;
```

Comment:
This is hard to read and easy to get wrong. Can we use parenthesis here to make it clear what the check is ?

https://3v4l.org/VoZEa

```suggestion
        return (! $platform instanceof MySQLPlatform) || $platform instanceof MySQL80Platform;
```

Reply:
I don't think we need these parenthesis.